library/include/Export.h

```diff
@@ -69,3 +69,17 @@ distribution.
 #else
     #define Wformat(type, fmtstr, vararg)
 #endif
+
+namespace  DFHack
+{
+    enum command_result
+    {
+        CR_LINK_FAILURE = -3,    // RPC call failed due to I/O or protocol error
+        CR_NEEDS_CONSOLE = -2,   // Attempt to call interactive command without console
+        CR_NOT_IMPLEMENTED = -1, // Command not implemented, or plugin not loaded
+        CR_OK = 0,               // Success
+        CR_FAILURE = 1,          // Failure
+        CR_WRONG_USAGE = 2,      // Wrong arguments or ui state
+        CR_NOT_FOUND = 3         // Target object not found (for RPC mainly)
+    };
+}
```

Comment:
Can this go in Core.h instead? Export.h is really just for macros that affect symbol visibility (well, and Wformat, which should probably move too).

Not familiar with what dependencies exist here. I know Export.h is convenient because it's always included.

Reply:
Hm, of the headers that use `command_result`, `RemoteClient.h` and `RemoteTools.h` are the only two that don't include `Core.h` already. But `Core.h` includes one of them, `RemoteClient.h`, so we can't introduce a circular include there.

I wouldn't be opposed to a separate `Command.h`. I guess there is also `Types.h`, which looks mostly outdated to me, and `Internal.h`, which at first glance seems _similar_ to `Export.h` but may cause some behavioral changes that we wouldn't want to force on _everything_ that includes DFHack headers (such as disabling warnings and forcing large file support on Linux. Worth noting that `Types.h` is currently only included by `.cpp` files within the `library` folder.)

Reply:
Note that Core.h included RemoteClient.h *because* of command_result. It might be ok to include the other way now

Reply:
Ah that would make sense. I didn't really dive into _why_ those headers were being included.

Reply:
done